processing/src/main/java/io/druid/query/groupby/having/DimensionSelectorHavingSpec.java

```diff
@@ -35,41 +37,55 @@
   private static final byte STRING_SEPARATOR = (byte) 0xFF;
   private final String dimension;
   private final String value;
+  private final ExtractionFn extractionFn;
 
   @JsonCreator
   public DimensionSelectorHavingSpec(
       @JsonProperty("dimension") String dimName,
-      @JsonProperty("value") String value
+      @JsonProperty("value") String value,
+      @JsonProperty("extractionFn") ExtractionFn extractionFn
   )
   {
-    this.dimension = Preconditions.checkNotNull(dimName, "Must have attribute 'dimension'");
+    dimension = Preconditions.checkNotNull(dimName, "Must have attribute 'dimension'");
     this.value = value;
+    if (extractionFn != null) {
+      this.extractionFn = extractionFn;
+    } else {
+      this.extractionFn = new IdentityExtractionFn(); 
```

Comment:
does this need to be a new instance? Can we make this a static constant somewhere?